testsuite/run_forge.sh

```diff
@@ -155,26 +147,44 @@ FORGE_END_TIME_MS="$(date '+%s')000"
 ES_END_TIME=$(TZ=UTC date +"%Y-%m-%dT%H:%M:%S.000Z")
 
 # parse the JSON report
+# also handle test report failure cases
 cat $FORGE_OUTPUT | awk '/====json-report-begin===/{f=1;next} /====json-report-end===/{f=0} f' >"${FORGE_REPORT}"
 # If no report was generated, fill with default report
 if [ ! -s "${FORGE_REPORT}" ]; then
     echo '{"text": "Forge test runner terminated"}' >"${FORGE_REPORT}"
+    FORGE_EXIT_CODE=1
+fi
+# If no report text was generated, fill with default text
+FORGE_REPORT_TXT=$(cat $FORGE_REPORT | jq -r .text)
+if [ -z "$FORGE_REPORT_TXT" ]; then
+    FORGE_REPORT_TXT="Forge report text empty. See test runner output."
+    FORGE_EXIT_CODE=1
 fi
 
 # print the Forge report
 cat $FORGE_REPORT
 
-# detect regressions
-# TODO(rustielin): do not block on perf regressions for now until Forge network performance stabilizes
+# detect regressions. TODO: do this in the Forge test runner itself since it's complex
+# if this script is not triggered in GHA, use a default value
+[ -z "$GITHUB_RUN_ID" ] && GITHUB_RUN_ID=0
+if echo $FORGE_CLUSTER_NAME | grep "forge"; then
+    ES_DEFAULT_INDEX=$DEVINFRA_ES_DEFAULT_INDEX
+    ES_BASE_URL=$DEVINFRA_ES_BASE_URL
+    GRAFANA_BASE_URL=$DEVINFRA_GRAFANA_BASE_URL
+else
+    ES_DEFAULT_INDEX=$INTERN_ES_DEFAULT_INDEX
+    ES_BASE_URL=$INTERN_ES_BASE_URL
+    GRAFANA_BASE_URL=$INTERN_GRAFANA_BASE_URL
+fi
 AVG_TPS=$(cat $FORGE_REPORT | grep -oE '[0-9]+ TPS' | awk '{print $1}')
 P99_LATENCY=$(cat $FORGE_REPORT | grep -oE '[0-9]+ ms p99 latency' | awk '{print $1}')
 if [ -n "$AVG_TPS" ]; then
     echo "AVG_TPS: ${AVG_TPS}"
     echo "forge_job_avg_tps {FORGE_CLUSTER_NAME=\"$FORGE_CLUSTER_NAME\",FORGE_NAMESPACE=\"$FORGE_NAMESPACE\",GITHUB_RUN_ID=\"$GITHUB_RUN_ID\"} $AVG_TPS" | curl -u "$PUSH_GATEWAY_USER:$PUSH_GATEWAY_PASSWORD" --data-binary @- ${PUSH_GATEWAY}/metrics/job/forge
     if [[ "$AVG_TPS" -lt "$TPS_THRESHOLD" ]]; then
         echo "(\!) AVG_TPS: ${avg_tps} < ${TPS_THRESHOLD} tps"
-	if [ "$FORGE_RUNNER_MODE" != "local" ]; then
-           FORGE_EXIT_CODE=1
+        if [ "$FORGE_RUNNER_MODE" != "local" ]; then
+            FORGE_EXIT_CODE=1
```

Comment:
Can we add the comment in the report that the test failed because of TPS/latency regression?

Reply:
done. lmk what you think.